### PR TITLE
Implement BNPL providers in classic checkout flow

### LINKED ISSUE #6119: Shortcode Checkout with BNPL providers
### Scenario 1: Display BNPL Providers at Checkout
**As a** buyer of a shop that has BNPL (Buy Now Pay Later) providers enabled,
**I want to** see which BNPL providers I can use on the checkout page,
**So that** I can choose one to pay for my order.

#### Acceptance Criteria

1. On the checkout page, a list of available BNPL providers should be displayed.
2. Each BNPL provider should have a clear and recognizable logo or name.
3. The buyer should be able to select one of the available BNPL providers to proceed with the payment process.
4. Each BNPL provider will appear in the list of available providers only if the purchase is compatible in terms of:
4.1. Customer location
4.2. Presentment currency
4.3. Recurring payments (a.k.a. subscriptions are not supported)
4.4. Country availability
4.5. Purchase limits
5. UPE and split UPE peculiarities should be taken into account.

Please check the following links for details:

- [Afterpay and Clearpay payments | Stripe Documentation](https://stripe.com/docs/payments/afterpay-clearpay)
- [Affirm payments | Stripe Documentation](https://stripe.com/docs/payments/affirm)

### Designs

TBD

### Dev notes
Based on our initial spikes, Split UPE and Legacy UPE work out of the box, please refer to screenshots below. Nonetheless, we should test both scenarios.

_Split UPE_
Screenshot 2023-05-05 at 19 33 56

_Legacy UPE_
Screenshot 2023-05-05 at 19 24 03

### Scenario 2:  Redirect to Confirmation Page after Successful BNPL Payment

Redirect to Confirmation Page after Successful BNPL Payment

**As a** buyer,
**I want to** be redirected to a confirmation page after successfully completing the payment process with a BNPL provider,
**So that** I know that my order is completed.

#### Acceptance Criteria

1. After the buyer successfully completes the payment process with the chosen BNPL provider, they should be redirected to a confirmation page.
2. The confirmation page should display relevant order information, such as order number, items, and total cost.
3. The page should also include a message confirming that the order has been completed.

### Designs

TBD

### Scenario 3: Redirect to Error Page after BNPL Payment Error

**As a** buyer,
**I want to** be redirected to an error page when there is an error in the payment process with a BNPL provider,
**So that** I know that my order is not completed and the reason why it was not completed.

#### Acceptance Criteria

1. If an error occurs during the payment process with the chosen BNPL provider, the buyer should be redirected to an error page.
2. The error page should display a clear and informative message explaining the reason for the payment failure.
3. The buyer should be given options to either try the payment process again or choose a different payment method.

### Designs

TBD

### Assumptions
We can use integration spikes as starting points and guidance on what needs to be accomplished.

### Dev notes

* We completed integration spikes: https://github.com/Automattic/woocommerce-payments-server/pull/3183 and https://github.com/Automattic/woocommerce-payments/pull/5910
### Additional context
* We need to test all possible breaking points in the complete flow. Merchant Checkout page->BNPL site->BNPL Approval / Reject->Redirect to Merchant store.
* We should only rely on webhooks sent by Stripe to update the status of the order.

### PR DESCRIPTION
**Pull request by KarlisJ**
Fixes #6119 

#### Changes proposed in this Pull Request

Provide option of Affirm and Afterpay at checkout for eligible merchants on a classic shortcode checkout page

#### Testing instructions

Please follow the [setup instruction of the base PR](https://github.com/Automattic/woocommerce-payments/pull/6492)

For each:
- legacy UPE 
- Split UPE

1. Enable Affirm and Afterpay in payment settings
2. Add items to cart (note that Affirm is available only for purchases over 50$)
3. Make a payment using new payment method
4. Watch for any errors in console or in the front-end flow.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: client/checkout/classic/upe-split.js [hunk 1]
```diff
@@ -539,6 +565,11 @@ jQuery( function ( $ ) {
 					},
 				},
 			};
+			if ( PAYMENT_METHOD_IDS.AFTERPAY_CLEARPAY === paymentMethodType ) {
```
Comment (KarlisJ) [inline client/checkout/classic/upe-split.js:569]: I have to yet add proper comment explaining this line

File: client/checkout/classic/upe.js [hunk 2]
```diff
@@ -554,6 +577,7 @@ jQuery( function ( $ ) {
 					payment_method_data: {
 						billing_details: getBillingDetails( formFields ),
 					},
+					shipping: getShippingDetails( formFields ),
```
Comment (KarlisJ) [inline client/checkout/classic/upe.js]: I have to yet add similar check as for split-upe and include only for Afterpay. On first glance, I don't know how to get the used payment method type in this function

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6544` or branch name `add/6119-bnpl-shortcode-checkout` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/6119-bnpl-shortcode-checkout) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: b9e0fed1810ca365b2a45ef02af1d2499ef21926
- Build time: 2023-06-16 12:59:49 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/checkout/classic/upe-split.js (hunk 1 above)
Comment (kalessil) [reply client/checkout/classic/upe-split.js:569]: I was about to request this.

File: client/checkout/classic/upe-split.js [hunk 3]
```diff
@@ -108,6 +111,29 @@ jQuery( function ( $ ) {
 		$form.removeClass( 'processing' ).unblock();
 	};
 
+	/**
+	 * Converts form fields object into Stripe `shipping` object.
+	 *
+	 * @param {Object} fields Object mapping checkout shippinh fields to values.
```
Comment (nagpai) [inline client/checkout/classic/upe-split.js]: Typo - <s>shippinh</s> :  shipping

File: client/checkout/classic/upe-split.js [hunk 4]
```diff
@@ -108,6 +111,29 @@ jQuery( function ( $ ) {
...
+	 * @return {Object} Stripe formatted `shpping` object.
```
Comment (nagpai) [inline client/checkout/classic/upe-split.js]: Maybe typo <s>`shpping`</s>  : ``shipping``

File: client/checkout/classic/upe.js (hunk 2 above)
Comment (kalessil) [reply client/checkout/classic/upe.js]: `const paymentMethodType = getSelectedUPEGatewayPaymentMethod();`?

**Review by kalessil — APPROVED** [review]
Pre-approving (+my comments). Code review only.

**Review by nagpai — CHANGES_REQUESTED** [review]
Two tiny typo errors. 

Taking a deeper look at the conditional bit for Afterpay and will comment separately.

File: client/checkout/classic/upe.js (hunk 2 above)
Comment (KarlisJ) [reply client/checkout/classic/upe.js]: @kalessil I'm getting weird `wc is not defined` error when using that method. By the look of it it seems the method works properly only for split UPE

Comment (kalessil) [reply client/checkout/classic/upe.js]: Oops, perhaps there is a similar one for legacy UPE?

Comment (KarlisJ) [reply client/checkout/classic/upe.js]: Found the way to do it in [this commit ](https://github.com/Automattic/woocommerce-payments/pull/6544/commits/0861a522415aa0906f37cd2d4476f4ba4dceddef)

File: client/checkout/classic/upe-split.js (hunk 3 above)
Comment (KarlisJ) [reply client/checkout/classic/upe-split.js]: Adressed

File: client/checkout/classic/upe.js [hunk 5]
```diff
@@ -145,11 +146,39 @@ jQuery( function ( $ ) {
 		$( '#wcpay_selected_upe_payment_type' ).val( paymentType );
 	};
 
+	// Set the selected UPE payment type field
```
Comment (nagpai) [inline client/checkout/classic/upe.js]: Super minor - I think this needs to be `Get`
```suggestion
	// Get the selected UPE payment type field
```

**Review by nagpai — APPROVED** [review]
LGTM! 

✅  - Tested Classic UPE checkout - Shipping details added only for Afterpay , and order flow works correctly
✅  - Tested Split UPE checkout -  Shipping details added only for Afterpay , and order flow works correctly

✅  Affirm does not appear for orders below threshold.

**Pull request merged**